src/itdelatrisu/opsu/beatmap/BeatmapSetList.java

```diff
@@ -100,13 +103,27 @@ private BeatmapSetList() {
 	 * This does not erase any parsed nodes.
 	 */
 	public void reset() {
+		resetFiltered();
 		nodes = groupNodes = BeatmapGroup.current().filter(parsedNodes);
 		expandedIndex = -1;
 		expandedStartNode = expandedEndNode = null;
 		lastQuery = null;
 	}
 
 	/**
+	 * Will clear the filter on any mapsets that matched the last search request
+	 */
+	private void resetFiltered(){
```

Comment:
For the sake of clarity, I think it's better to just call `clearFilter()` on everything in `nodes`, and not bother keeping the extra `lastSearchNodes` reference.  It's not noticeably helping performance, and I don't want this class getting much more complex (it's already pretty confusing).

Alternatively, since `lastSearchNodes` is just pointing to `nodes` anyway, why not use a boolean flag instead?